storage/core/src/main/java/com/hortonworks/streamline/storage/filestorage/DbFileStorage.java

```diff
@@ -0,0 +1,93 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.streamline.storage.filestorage;
+
+import com.hortonworks.streamline.common.util.FileStorage;
+import com.hortonworks.streamline.storage.StorageManager;
+import com.hortonworks.streamline.storage.StorageManagerAware;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * DB based file storage that stores the file as a blob.
+ */
+public class DbFileStorage implements FileStorage, StorageManagerAware {
+    private static final Logger LOG = LoggerFactory.getLogger(DbFileStorage.class);
+    private StorageManager dao;
+
+    @Override
+    public void init(Map<String, String> config) {
+    }
+
+    @Override
+    public String uploadFile(InputStream inputStream, String name) throws IOException {
+        LOG.debug("Uploading '{}'", name);
+        long start = System.nanoTime();
+        Optional<FileBlob> existing = get(name);
+        if (existing.isPresent()) {
+            LOG.debug("Updating existing file '{}'", name);
+            FileBlob updated = new FileBlob(existing.get());
+            updated.setData(inputStream);
+            updated.setVersion(existing.get().getVersion() + 1);
+            updated.setTimestamp(System.currentTimeMillis());
+            dao.update(updated);
+        } else {
+            LOG.debug("Adding new file '{}'", name);
+            FileBlob fileBlob = new FileBlob();
+            fileBlob.setName(name);
+            fileBlob.setTimestamp(System.currentTimeMillis());
+            fileBlob.setVersion(0L);
+            fileBlob.setData(inputStream);
+            dao.add(fileBlob);
+        }
+        LOG.debug("Uploaded '{}' in '{}' milliseconds", name, (System.nanoTime() - start)/1000000);
+        return name;
+    }
+
+    @Override
+    public InputStream downloadFile(String name) throws IOException {
+        LOG.debug("Downloading file '{}'", name);
+        FileBlob res = get(name)
+                .orElseThrow(() -> new IOException("Not able to get file blob with name : " + name));
+        return res.getData();
+    }
+
+    @Override
+    public boolean deleteFile(String name) throws IOException {
+        LOG.debug("Deleting file '{}'", name);
+        return dao.remove(FileBlob.getStorableKey(name)) != null;
+    }
+
+    @Override
+    public boolean exists(String name) {
```

Comment:
I think ideally the other methods should be renamed e.g. `upload` instead of `uploadFile` (since its already defined in the context of FileStorage and so implies that we are dealing with files). I will rename the other methods.